contracts/ColonyStorage.sol

```diff
@@ -125,13 +125,50 @@ contract ColonyStorage is CommonStorage, ColonyDataTypes, DSMath {
     _;
   }
 
-  modifier isAdmin(address _user) {
-    require(ColonyAuthority(address(authority)).hasUserRole(_user, uint8(ColonyRole.Admin)), "colony-not-admin");
+  modifier isAdmin(uint256 _parentDomainId, uint256 _childIndex, uint256 _id, address _user) {
+    require(ColonyAuthority(address(authority)).hasUserRole(_user, _parentDomainId, uint8(ColonyRole.Administration)), "colony-not-admin");
+    require(isValidDomainProof(_parentDomainId, tasks[_id].domainId, _childIndex), "colony-invalid-domain-proof");
     _;
   }
 
   modifier self() {
     require(address(this) == msg.sender, "colony-not-self");
     _;
   }
+
+  modifier authDomain(uint256 _parentDomainId, uint256 _childDomainId, uint256 _childIndex) {
+    require(isAuthorized(msg.sender, _parentDomainId, msg.sig), "ds-auth-unauthorized");
+    require(isValidDomainProof(_parentDomainId, _childDomainId, _childIndex), "ds-auth-invalid-domain-proof");
+    if (canCallBecauseArchitect(msg.sender, _parentDomainId, msg.sig)) {
+      require(_parentDomainId != _childDomainId, "ds-auth-only-authorized-in-child-domain");
+    }
+    _;
+  }
+
+  modifier auth() {
+    require(isAuthorized(msg.sender, msg.sig), "ds-auth-unauthorized");
+    _;
+  }
```

Comment:
Hmm, I assumed we had to copy `auth` here since we override `isAuthorized`, but maybe that's not the case.

I think keeping `auth` is good, since it's more clear in the case of globally-`auth`ed functions like the `root` and `recovery` functions. For those we'd have to do something like `authDomain(1, 0, 0)` and not actually pass in any arguments, which to my mind is more confusing than just `auth`ing those functions. I think the distinction that `auth` refers to global permissions vs `authDomain` which refers to domains helps communicate whats going on.

Reply:
But in the end `auth` here is in fact referring to domain permissions where the domain is the root domain id 1 right?

Reply:
Yup, but transparently, and only in ColonyStorage do you realize that. From the perspective of the other contracts that doesn't matter, which I think is valuable.

Reply:
However I was able to remove the override of `isAuthorized` which I noticed was unnecessary.